Sources/SwiftKueryPostgreSQL/PostgreSQLConnection.swift

```diff
@@ -491,4 +504,28 @@ public class PostgreSQLConnection: Connection {
         PQclear(result)
         onCompletion(.successNoData)
     }
+
+    private func buildQuery(_ query: Query) throws  -> String {
+      var postgresQuery = try query.build(queryBuilder: queryBuilder)
+      if (query is Insert) {
+
+        guard let insertQuery = query as? Insert,
+              let returnIDFlag = insertQuery.returnID
+        else {
+          throw QueryError.downcastError("Error downcasting Query to Insert")
+        }
+
+        let columns = insertQuery.table.columns.filter { $0.isPrimaryKey && $0.autoIncrement }
+
+        if (returnIDFlag && insertQuery.suffix == nil && columns.count == 1) {
+           let insertQueryReturnID = insertQuery.suffix("Returning " + columns[0].name)
+           postgresQuery = try insertQueryReturnID.build(queryBuilder: queryBuilder)
+        }
+
+        if (columns.count > 1) {
+          throw QueryError.databaseError("2 or more columns set as primary key and auto increment could not decide which one to return")
+        }
+      }
+      return postgresQuery
+    }
```

Comment:
It feels like we shouldn't do any of this if `returnID` is `false`. Also, if a `suffix` is already set then we ignore the request to return an ID. Perhaps we should throw an error and/or log something in that case.

Reply:
Done